core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceTest.java

```diff
@@ -315,13 +319,76 @@ public String glob2(ServiceRequestContext ctx) {
 
         @Get
         @Path("regex:^/regex/(?<path>.*)$")
-        public String regex(ServiceRequestContext ctx, @PathParam("path") String path) {
+        public String regex(ServiceRequestContext ctx, @Param("path") String path) {
             return "regex:" + ctx.path() + ':' + path;
         }
     }
 
+    @Converter(target = String.class, value = UnformattedStringConverter.class)
+    public static class MyAnnotatedService7 {
+        @Get
+        @Path("/param/get")
+        public String paramGet(RequestContext ctx,
+                               @Param("username") String username,
+                               @Param("password") String password) {
+            validateContext(ctx);
+            return username + "/" + password;
+        }
+
+        @Post
+        @Path("/param/post")
+        public String paramPost(RequestContext ctx,
+                                @Param("username") String username,
+                                @Param("password") String password) {
+            validateContext(ctx);
+            return username + "/" + password;
+        }
+
+        @Get
+        @Path("/map/get")
+        public String mapGet(RequestContext ctx, ParamMap parameters) {
+            validateContext(ctx);
+            return parameters.getFirst("username") + "/" + parameters.getFirst("password");
+        }
+
+        @Post
+        @Path("/map/post")
+        public String mapPost(RequestContext ctx, ParamMap parameters) {
+            validateContext(ctx);
+            return parameters.getFirst("username") + "/" + parameters.getFirst("password");
+        }
+
+        @Get
+        @Path("/param/default1")
+        public String paramDefault1(RequestContext ctx,
+                               @Param(value = "username", defaultValue = "hello") String username,
+                               @Param(value = "password", required = false, defaultValue = "world")
```

Comment:
I tried `@Optional`. It seems that it's not confused to `java.util.Optional` because this is an annotation. 

Also, I almost finished the fix specifying path pattern on HTTP method annotations such as `@Get` and `@Post`. I will add some more tests, then let you know.